src/TapeChart.js

```diff
@@ -5,47 +5,55 @@ class TapeChart {
     this.bookings = bookings;
   }
 
-getAvailableRoomsByDate(date) {
+  getAvailableRooms(date, roomType) {
     let availableByDate = this.rooms.reduce((acc, room) => {
       let bookingsByRoom = this.bookings.filter(booking =>
         Number(booking.roomNumber) === room.number);
       if (!bookingsByRoom.some(booking => booking.date === date)) {
         acc.push(room);
       }
-        return acc;
-    }, [])
-  return availableByDate;
-}
+      return acc;
+    }, []);
+    if (roomType && roomType !== 'any') {
+      return this.filterByRoomType(availableByDate, roomType);
+    } else {
+      return availableByDate;
+    }
+  }
 
-getOccupancy(date) {
-  return 100 - (this.getAvailableRoomsByDate(date).length
-    / this.rooms.length * 100);
-}
+  filterByRoomType(roomList, roomType) {
+    return roomList.filter(room => room.roomType === roomType)
+  }
 
-filterBookingsByMetric(metric, spec) {
-  return this.bookings.filter(booking => booking[metric] === spec);
-}
+  getOccupancy(date) {
+    return 100 - (this.getAvailableRooms(date).length
+      / this.rooms.length * 100);
+  }
 
-getDaysRevenue(metric, spec) {
-  let bookingsByDate = this.filterBookingsByMetric(metric, spec);
-  return Number(bookingsByDate.reduce((acc, booking) => {
-    this.rooms.forEach(room => {
-      if (room.number === Number(booking.roomNumber)) {
-        acc += room.costPerNight;
-      }
-    })
-    return acc;
-  }, 0).toFixed(2));
-}
+  filterBookingsByMetric(metric, spec) {
+    return this.bookings.filter(booking => booking[metric] === spec);
+  }
 
-filterAvailableRoomsByFeature(date, feature, spec) {
-  let availableRoomsByDate = this.getAvailableRoomsByDate(date);
-  return availableRoomsByDate.filter(room => room[feature] === spec);
-}
+  getDaysRevenue(metric, spec) {
+    let bookingsByDate = this.filterBookingsByMetric(metric, spec);
```

Comment:
getAvailableRooms(date) would this work better for this line? I might just need to see how you are calling this to understand?

Reply:
`getAvailableRooms()` would give me the rooms that are *not* booked for a date.

For this I need to access the opposite.  I need all the bookings/rooms that ARE booked for that date.